packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js

```diff
@@ -148,6 +154,53 @@ function InstalledFonts() {
 		refreshLibrary();
 	}, [] );
 
+	// Get activated fonts count.
+	const activeFontsCount = libraryFontSelected
+		? getFontFacesActivated(
+				libraryFontSelected.slug,
+				libraryFontSelected.source
+		  ).length
+		: 0;
+
+	// Check if any fonts are selected.
+	const isIndeterminate =
+		activeFontsCount > 0 &&
+		activeFontsCount !== libraryFontSelected?.fontFace?.length;
+
+	// Check if all fonts are selected.
+	const isSelectAllChecked =
+		activeFontsCount === libraryFontSelected?.fontFace?.length;
+
+	// Toggle select all fonts.
+	const toggleSelectAll = () => {
+		const initialFonts = fontFamilies?.[ libraryFontSelected.source ] ?? [];
+		const deactivateFonts = isSelectAllChecked || isIndeterminate;
```

Comment:
```suggestion
		const deactivateFonts = isSelectAllChecked;
```

The current condition is that if some fonts are selected and you check "Select All", all fonts will be unchecked.

It would be good to align this behavior with the Block Lock UI, i.e. if some fonts are selected and you check "Select All", all fonts will be checked.

https://github.com/user-attachments/assets/eedca853-bf2c-4dad-850d-ed4fb32c86ad


However, making this change will log errors to the browser console. We may need to adjust some logic to go along with this change.

```
Warning: Encountered two children with the same key, `heading`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version
```